qa/standalone/test_ceph_daemon/test_ceph_daemon_tar.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash -ex
+
+#
+# Generates a tarball for use with
+# the standalone `ceph-daemon` tests.
+#
+
+DIRS=(
+   "/etc/ceph"
+   "/var/lib/ceph"
+   "/var/log/ceph"
+)
+
+SCRIPT_NAME=$(basename ${BASH_SOURCE[0]})
+
+TMPDIR=$(mktemp -d ${SCRIPT_NAME%.*}.XXX)
+trap "rm -rf $TMPDIR" EXIT
+
+TARBALL="$1"
+if [ -z "$TARBALL" ]; then
+    echo "tarball not specified" 1>&2
+    exit 1
+fi
+
+# `systemctl stop` all services
+systemctl list-units | grep ceph
+systemctl stop ceph.target
+
+# Copy the required dirs
+for dir in "${DIRS[@]}"; do
+    cp -a --parents --sparse=always $dir $TMPDIR
+done
+
+# Convert block dev(s) into a sparse file
+SYMLINKS=$(find $TMPDIR -type l)
+for symlink in $SYMLINKS; do
+    file -L $symlink |  grep "block special" || exit 1
+    dd if=$symlink of=$symlink.dev conv=sparse
+    rm -f $symlink && ln -s $(basename $symlink.dev) $symlink
+done
+
+# Output the tarball
+tar czvSpf deepsea.tgz -C $TMPDIR .
```

Comment:
> qa/standalone/test_ceph_daemon/deepsea.tgz

can we avoid adding binary to the repo?

Reply:
that's going to be kind of similar to the corpus. It's supposed to be a set of directory layouts generated by e.g. ceph-deploy that needs to be adoptable from ceph-daemon.

Reply:
no, i don't think it's like the corpus. the purpose of having ceph corpus repo is to keep track of different versions of encoded ceph data structures. it's difficult to setup and tear down different versions of ceph cluster of different versions just for generating and post processing them for a ready-to-use corpus before testing forward/backward compatibility.

if we are able to generate the disk layout using ceph-deploy, why not do this with ceph-deploy, and teardown the cluster, then let ceph-daemon to reconstruct the cluster by adopting the tarball?

Reply:
I think this a bit different because some of these tools' behavior changes over time and it's really helpful to capture the way something was deployed by some tool at some point in time.  Also, some of the tools may stop working entirely.  A filestore OSD deployed by ceph-disk is a perfect example.  Or perhaps early bluestore deployments from ceph-disk.

I'm fine with switching the location for these to some other static location... maybe a submodule? Or we can dump them in a directory on qa.ceph.com or something.  A submodule is probably easier/simpler to maintain/manage, though!

Reply:
The tgz file has been moved to the corpus:
https://github.com/ceph/ceph-daemon-adoption-corpus/commit/2c16144aff4cf7290c02923e44945dc54e38d5ff

I tried to follow the pattern used by the [`ceph-object-corpus`](https://github.com/ceph/ceph-object-corpus/tree/master/archive) where the files reside in a structure `archive/<ceph_version>/<any_name>.tgz`

But I'm happy to re-arrange or rename if there is a strong preference otherwise.

Reply:
@mgfritch thanks!